pkg/proxy/userspace/proxier.go

```diff
@@ -461,16 +467,20 @@ func (proxier *Proxier) addServiceOnPortInternal(service proxy.ServicePortName,
 		protocol:            protocol,
 		socket:              sock,
 		sessionAffinityType: v1.ServiceAffinityNone, // default
+		started:             &sync.WaitGroup{},
+		finished:            &sync.WaitGroup{},
 	}
+	si.started.Add(1)
+	si.finished.Add(1)
 	proxier.serviceMap[service] = si
 
 	klog.V(2).Infof("Proxying for service %q on %s port %d", service, protocol, portNum)
-	go func(service proxy.ServicePortName, proxier *Proxier) {
+	go func() {
 		defer runtime.HandleCrash()
-		atomic.AddInt32(&proxier.numProxyLoops, 1)
+		defer si.finished.Done()
+		si.started.Done()
```

Comment:
what is blocking on `si.started.Wait()` intended to indicate? it doesn't mean sock.ProxyLoop() has been called yet... it means it's *about* to be called

Reply:
@liggitt nor did waitForProxyLoop(,.., 1) mean it was started yet either... I think we just care that the goroutine is started, since sometimes that takes a while.

Reply:
> I think we just care that the goroutine is started, since sometimes that takes a while

I can't think of anything that would wait on si.started that wouldn't be flaky... the listener is already established before addServiceOnPortInternal returns, so there's no need to wait on si.started before sending requests to it... I think we should drop si.started and simplify `finished` to something like a channel we close